content/5_Plat/RURQ.mdx

```diff
@@ -221,9 +221,165 @@ queries.
 	</Resource>
 </Resources>
 
-### Implementation
+### Solution - Range Updates & Sums
 
-<IncompleteSection />
+<Spoiler title="Solution">
+
+This question asks you to support the following types of queries. 
+
+-	Add a value to all elements within the range $[a,b]$.
+
+-	Set all values within the range $[a,b]$ to a certain value. 
+
+-	Find the sum of all values in range $[a,b]$.
+
+Consider the first two types of queries, a lazy tag will be created in each node of the tree for each type. In this solution, $lzAdd$ will represent the lazy tag for the range add query and $lzSet$ will represent the lazy tag for the range set query. 
+
+Given the two different update queries, a total of four different situations might take place as a result of any update. 
+
+-	Range add $lzAdd$ when $lzSet$ equals 0: Simply add the new value to the pre-existing value. 
+
+-	Range add $lzAdd$ when $lzSet$ doesn't equal 0: Add the new value to $lzSet$ and clear $lzAdd$. 
+
+-	Range set when $lzAdd$ equals 0: Simply update the $lzSet$ value. 
+
+-	Range set when $lzAdd$ doesn't equal 0: Again simply update the $lzSet$ value since a set update will override all previous add update. 
+
+Given the mechanics behind the ```pushdown``` function, all you need is a regular range sum segment tree to complete the question. 
+
+```cpp
+
+#include <iostream>
+#define int long long 
```

Comment:
> #define int long long

why is this still here :/

Reply:
I replaced it with #define ll long long

Reply:
As mentioned [here](https://usaco.guide/general/generic?lang=cpp#macros), `using ll = long long;` is preferred.

Also, please don't just make everything `ll` ...

Reply:
Fixed. On a side note would making everything long long be considered as a bad habit that I should avoid if I am sure that MLE is not going to be an issue for this particular problem.

Reply:
well I guess it isn't that bad, but if you use different data types for the indices vs the values it's easy to see what you need to change if you want to use a different value type.

btw - the code now fails ...
<img width="508" alt="Screen Shot 2021-03-30 at 10 50 12 AM" src="https://user-images.githubusercontent.com/21228024/113009178-b9265480-9145-11eb-8c71-2ab96762b1f6.png">

Reply:
yea I realized that I overchanged long long variables back to int and caused overflow... I think thats how my habit of making all integers long long started. I tested again and it is AC this time.